index.js

```diff
@@ -60,6 +61,10 @@ module.exports = {
               return res.send({ bearer: bearer });
             }
           });
+          async function emitEventBeforeLogin(callback) {
+            await self.emit('beforeLogin', req);
```

Comment:
This may throw an exception, in which case it must be caught and passed to the callback as the first argument.